composition/string_fragment.go

```diff
@@ -5,13 +5,29 @@ import (
 )
 
 // StringFragment is a simple template based representation of a fragment.
-type StringFragment string
+type StringFragment struct {
```

Comment:
The struct type is not necessary (in my opinion). You can add everything you need to the string.

type bla string{
stuff
}

@smancke your opinion? I think it generates overhead